### PR TITLE
Add alifestd_coarsen_dilate_asexual phylogeny coarsening function

### PR DESCRIPTION
**Pull request by mmore500**
## Summary

This PR adds a new phylogeny coarsening function `alifestd_coarsen_dilate_asexual` that collapses inner nodes within dilation windows, along with a Polars-based implementation and comprehensive test coverage.

## Key Changes

- **New core function**: `alifestd_coarsen_dilate_asexual()` in pandas that coarsens phylogenies by collapsing inner (non-leaf) nodes whose criterion values fall within the same dilation window `[n, n + dilation)` where `n % dilation == 0`
  - Snaps inner node criterion values to the nearest lower boundary
  - Merges consecutive inner nodes that land on the same boundary
  - Preserves all leaf nodes unchanged
  - Guarantees MRCA shifts backward by at most `dilation` units

- **Polars implementation**: `alifestd_coarsen_dilate_polars()` providing equivalent functionality for Polars DataFrames and LazyFrames
  - Reuses the core `_coarsen_dilate_impl()` numba-jitted algorithm
  - Supports both eager and lazy evaluation

- **Core algorithm**: `_coarsen_dilate_impl()` - a numba-jitted single-pass implementation that:
  - Computes leaf status by checking which nodes are referenced as ancestors
  - Snaps inner node criterion values to dilation boundaries
  - Merges nodes landing on the same boundary by updating representative mappings
  - Returns updated ancestor IDs, criterion values, and a keep mask

- **CLI entrypoints**: Both pandas and Polars implementations include command-line interfaces supporting CSV/Parquet I/O with configurable dilation and criterion parameters

- **Comprehensive test suite** (466 tests for pandas, 317 for Polars):
  - Edge cases: empty phylogenies, singletons, chains, branching trees, star topologies
  - Dilation variations: dilation=1, dilation=2, large dilation values
  - Float and integer criterion values
  - Custom criterion columns
  - Topological invariants: leaf preservation, MRCA shift bounds, inner node count monotonicity
  - Cross-implementation validation (pandas vs Polars)

## Implementation Details

- Requires topologically sorted input with contiguous IDs
- Supports custom criterion columns (default: `origin_time`)
- Handles `ancestor_list` column regeneration in pandas implementation
- Includes topological sensitivity warnings via decorator
- Optional in-place mutation support in pandas version
- Comprehensive input validation (positive dilation, criterion column existence)

https://claude.ai/code/session_011bkQ2FMFEaLgiV6nU4USPJ

### CODE REVIEW

**Review by mmore500 — COMMENTED** [review]
changes requested

File: hstrat/_auxiliary_lib/_alifestd_coarsen_dilate_asexual.py [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+    )
+    return parser
+
+
+if __name__ == "__main__":
+    begin_prod_logging()
+
+    parser = _create_parser()
+    args, remaining = parser.parse_known_args()
+    if (
```
Comment (mmore500) [inline hstrat/_auxiliary_lib/_alifestd_coarsen_dilate_asexual.py]: this should be handled by argparse, no?

**Review by mmore500 — COMMENTED** [review]
changes requested

**Review by mmore500 — COMMENTED** [review]
lgtm
